piksi_tools/settings.py

```diff
@@ -163,58 +159,63 @@ def read(self, section, setting, retries=DEFAULT_READ_RETRIES, verbose=False):
             raise RuntimeError(("Unable to read setting \"{}\" in section \"{}\" after {} attempts. "
                                 "Setting may not exist.".format(setting, section, retries)))
 
-    def write(self, section, setting, value, write_retries=DEFAULT_WRITE_RETRIES,
-              confirm_retries=DEFAULT_CONFIRM_RETRIES, verbose=False):
+    def write(self, section, setting, value, verbose=False):
         """Write setting by name and confirm set
 
-        Notes: Caller does not raise an exception, but RuntimeError may be raised when write unsuccessful
-
         Args:
             section(str): string of section name
             setting(str): string of setting name
             value(str): value to set
             verbose(bool): Echo settings to sdout
-
+        Raises:
+            KeyError if the section/setting does not exist
+            ReadOnlySettingError if the section/setting is read-only
+            RuntimeError if the write was otherwise unsuccessful
         Returns:
             None
-
         """
-
-        attempts = 0
-        while (attempts < write_retries):
+        if verbose:
+            print("Attempting to write: section={}|setting={}|value={}".format(section, setting, value))
+
+        key = b'%s\0%s\0' % (section.encode(KEY_ENCODING), setting.encode(KEY_ENCODING))
+        request = key + b'%s\0' % value.encode(VALUE_ENCODING)
+
+        self.link(MsgSettingsWrite(setting=request))
+        reply = self.link.wait(SBP_MSG_SETTINGS_WRITE_RESP, timeout=self.timeout)
+
+        if reply is None:
+            raise RuntimeError("Settings write request {}/{} timed out"
+                               .format(section, setting))
+        # NOTE: reply payload is not set correctly when status is 2
```

Comment:
Check if the bug is still active with the latest firmware/buildroot master tip.

Reply:
If it is, assign the issue to me.

Reply:
seems to work now, removing the workaround